spinoso-string/src/enc/utf8/mod.rs

```diff
@@ -809,10 +809,37 @@ impl Utf8String {
     #[inline]
     #[must_use]
     pub fn index(&self, needle: &[u8], offset: usize) -> Option<usize> {
-        for byte_pos in self.find_iter(needle) {
-            let char_pos = self[..byte_pos].chars().count();
-            if char_pos >= offset {
-                return Some(char_pos);
+        // Decode needle
+        let needle_chars = match must_decode_utf8(needle) {
+            Some(chars) => chars,
+            // Needle containing any invalid UTF-8 should never match in MRI
+            None => return None,
```

Comment:
According to `irb`:
<img width="1457" alt="image" src="https://github.com/artichoke/artichoke/assets/12410942/3be0ad1e-695f-40d8-86f8-e0cdf773c16a">